sysdeps/managarm/generic/file.cpp

```diff
@@ -1142,7 +1142,11 @@ int sys_msg_send(int sockfd, const struct msghdr *hdr, int flags, ssize_t *lengt
 }
 
 int sys_msg_recv(int sockfd, struct msghdr *hdr, int flags, ssize_t *length) {
-	__ensure(hdr->msg_iovlen);
+	if(!hdr->msg_iovlen) {
+		errno = EMSGSIZE;
+		return -1;
```

Comment:
We should return the error in a sysdep, not set errno.

Reply:
Addressed in 1dc288a.